src/components/destroy-at-extreme-distances.js

```diff
@@ -8,9 +8,14 @@ AFRAME.registerComponent("destroy-at-extreme-distances", {
     zMax: { default: 1000 }
   },
 
+  init() {
+    this.el.sceneEl.systems["components-queue"].register(this, "media-components");
+  },
+
   tick: (function() {
     const pos = new THREE.Vector3();
     return function() {
+      if (!this.el.sceneEl.systems["components-queue"].shouldTick(this)) return;
```

Comment:
Instead of this I would just remove the `tick` method entirely have have the `component-queue` system call a different method (either passed into `register` or just with a specific name). This way aframe never registers this component as having a behavior and avoiding the extra work and code noise of doing an early out form `tick`